src/main/java/com/google/firebase/messaging/AndroidNotification.java

```diff
@@ -309,6 +414,185 @@ public Builder setImage(String imageUrl) {
+        } else {
+          list.add(String.format("%ds", seconds));
+        }   
+      }
+      this.vibrateTimings = ImmutableList.copyOf(list);  
+      return this;
+    }
+
+    /**
+     * Sets the whether to use the default vibration timings. If set to true, use the Android 
+     * framework's default vibrate pattern for the notification. Default values are specified 
+     * in config.xml. If default_vibrate_timings is set to true and vibrate_timings is also 
+     * set, the default value is used instead of the user-specified vibrate_timings.
+     *
+     * @param defaultVibrateTimings The flag incicating whether to use the default vibration timings
+     * @return This builder.
+     */
+    public Builder setDefaultVibrateTimings(boolean defaultVibrateTimings) {
+      this.defaultVibrateTimings = defaultVibrateTimings;
+      return this;
+    }
+
+    /**
+     * Sets the whether to use the default sound. If set to true, use the Android framework's 
+     * default sound for the notification. Default values are specified in config.xml.
+     *
+     * @param defaultSound The flag incicating whether to use the default sound
+     * @return This builder.
+     */
+    public Builder setDefaultSound(boolean defaultSound) {
+      this.defaultSound = defaultSound;
+      return this;
+    }
+
+    /**
+     * Sets the settings to control the notification's LED blinking rate and color if LED is 
+     * available on the device. The total blinking time is controlled by the OS.
+     *
+     * @param lightSettings The light settings to use
+     * @return This builder.
+     */
+    public Builder setLightSettings(LightSettings lightSettings) {
+      this.lightSettings = lightSettings;
+      return this;
+    }
+
+    /**
+     * Sets the whether to use the default light settings. If set to true, use the Android
+     * framework's default LED light settings for the notification. Default values are 
+     * specified in config.xml. If default_light_settings is set to true and 
+     * light_settings is also set, the user-specified light_settings is used instead of 
+     * the default value.
+     *
+     * @param defaultLightSettings The flag incicating whether to use the default light 
+     *     settings
+     * @return This builder.
+     */
+    public Builder setDefaultLightSettings(boolean defaultLightSettings) {
+      this.defaultLightSettings = defaultLightSettings;
+      return this;
+    }
+
+    /**
+     * Sets the visibility of this notification.
+     *
+     * @param visibility The visibility value. one of the values in {PRIVATE, PUBLIC, SECRET}
+     * @return This builder.
+     */
+    public Builder setVisibility(Visibility visibility) {
+      this.visibility = visibility;
+      return this;
+    }
+
+    /**
+     * Sets the number of items this notification represents. May be displayed as a badge 
+     * count for launchers that support badging.See Notification Badge. For example, this 
```

Comment:
Missing space I think.

Reply:
This should be a link to a different page https://developer.android.com/training/notify-user/badges. Decided to just remove the sentence.